### PR TITLE
Fix skipped gradient update of the last step when using gradient accumulation.

### PR DESCRIPTION
**Pull request by richardSHkim**
# What does this PR do?

This PR fixes skipped gradient update of the last step when using gradient accumulation in trainer.py.

I'm not sure what is the meaning of ['step is always smaller than gradient_accumulation_steps'](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L1995), but [steps_in_epoch <= args.gradient_accumulation_steps](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L1989) looks like always `False`.
Thus, the update of the last iteration is skipped when gradient_accumulation_step > 1.

This also leads to different total iterations (-1 iteration per epoch).

It might be negligible, and the skipped gradients are used in next iteration, but I believe the gradient update behavior should be the same whether or not using gradient accumulation if the total batch size is the same.

I have tested with [example script](https://github.com/huggingface/transformers/blob/main/examples/pytorch/image-classification/run_image_classification.py) and the comparison results are [here](https://wandb.ai/kshtrue95/fix_grad_accumulation/workspace?nw=nwuserkshtrue95).

accelerator.accumulate() context manager now can handle the last step of data loader, so I'm trying to use it.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@muellerzr @pacman100

### CODE REVIEW

**Comment by richardSHkim** [comment]
I found this implementation was intended in (PR#24415)[https://github.com/huggingface/transformers/pull/24415], so close this PR.
